Code-4-Community/speak-for-the-trees-frontend
# Optimize for landing page mobile

## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/702429655/pulses/886345302)

What benefit does this bring to the end user? Or, what benefit does this bring to developers working in the codebase?

Makes the logged-in landing page easier to navigate on mobile devices/narrow windows.  `useWindowDimensions` can be used to make other components responsive to browser dimensions.

## This PR

Describe the changes required and any implementation choices you made to give context to reviewers.

Created `useWindowDimensions` to get and use the `windowDimensions` state when rendering components
- `windowDimensions` contains the browser's `width`, `height`, and `windowType`
- `windowType` is determined by `width`:
  - `< 650` => `'MOBILE'`
  - `[650, 900)` => `'TABLET'`
  - `[900, 1150)` => `'NARROW'`
  - `> 1150` => `'DESKTOP'`

Created link carousel component using Ant.D Carousel component
- number of slides per page in carousel depends on window dimensions

Updated PageHeader and HomeContainer components so that top margin decreases for mobile

## Screenshots

Provide screenshots of any new components, styling changes, or pages.
Desktop, Mobile
<img width="420" alt="Desktop" src="https://user-images.githubusercontent.com/22990100/101569153-eb972080-39a1-11eb-8112-e5bb9abd8d6b.png"> <img width="109" alt="Mobile" src="https://user-images.githubusercontent.com/22990100/101569189-fa7dd300-39a1-11eb-82b4-1bfe820b1fcf.png">
Narrow, Tablet
<img width="185" alt="NarrowDesktop" src="https://user-images.githubusercontent.com/22990100/101569166-ef2aa780-39a1-11eb-9a07-af524e3857bd.png"><img width="160" alt="Tablet" src="https://user-images.githubusercontent.com/22990100/101569176-f782e280-39a1-11eb-9c28-4ad76e8f60b9.png">

## Verification Steps

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves.

Visually confirmed the layout updated when resizing the browser, also used Device Mode in Chrome Dev Tools to make sure the layout made sense for mobile.
